core/src/components/button/button.tsx

```diff
@@ -233,20 +233,15 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
   }
 
   /**
-   * Disable the "xsmall" and "xlarge" sizes if the theme is "ios" or "md"
+   * Set the size based if the button is inside of an item, the default size is "small"
    */
   private getSize(): string | undefined {
-    const theme = getIonTheme(this);
     const { size } = this;
 
     if (size === undefined && this.inItem) {
       return 'small';
     }
```

Comment:
```suggestion
    if (size === undefined) {
      return this.inItem ? 'small' : 'medium';
    }
```

If we go with adding `medium` above

Reply:
Commited 👍